app/api/custom/orders.py

```diff
@@ -145,3 +147,33 @@ def create_order():
     order.populate_and_save()
 
     return OrderSchema().dump(order)
+
+
+@order_blueprint.route('/attendees/<int:attendee_id>.pdf')
+@jwt_required
+def ticket_attendee_pdf(attendee_id):
+    ticket_holder = TicketHolder.query.get(attendee_id)
+    if ticket_holder is None:
+        raise NotFoundError(
+            {'source': ''}, 'This attendee is not associated with any ticket'
+        )
+
+    if not (
+        current_user.is_staff
+        or ticket_holder.order.user_id == current_user.id
+        or ticket_holder.user == current_user
+        or has_access(
+            'is_coorganizer_or_user_itself',
+            event_id=ticket_holder.event_id,
+            user_id=ticket_holder.user.id,
+        )
+    ):
+
+        raise ForbiddenError({'source': ''}, 'Unauthorized Access')
+    else:
+        file_path = ticket_holder.pdf_url_path
+        if os.path.exists(file_path):
+            return send_from_directory('../', file_path, as_attachment=True)
+        else:
+            create_pdf_tickets_for_holder(ticket_holder.order)
+            return send_from_directory('../', file_path, as_attachment=True)
```

Comment:
```suggestion
        if not os.path.isfile(file_path):
            create_pdf_tickets_for_holder(ticket_holder.order)
        return send_from_directory('../', file_path, as_attachment=True)
```

Think of the simplest way to achieve a solution which minimizes duplication

Reply:
And this change needs to be in other endpoint as well